### DISCUSSION TITLE
better support for infinite quotas

### DISCUSSION THREAD

**Discussion by DHD-Harper** [pretix/pretix#4574] [Ideas] [answered]
It would be desirable if pretix better supports infinite quotas.

A reservation period must necessarily be entered. However, this is irrelevant here. If 0 is entered, the cart is immediately displayed as expired.
A maximum number is also enforced for add-ons and a note on the maximum number that can be ordered is displayed in the frontend. This is unhelpful if the customer is theoretically allowed to order infinitely.

It would be nice to make those options optional.

**Comment by raphaelm** [comment, marked as answer]
> A reservation period must necessarily be entered. However, this is irrelevant here.

It's not that simple. Cart expiry is also important with unlimited quotas. For example:

- A voucher is reserved until the cart expires
- Seats are reserved until the cart expires
- Prices are guaranteed until the cart expires, e.g. in case there is a price change while someone is buying
- We need to clear carts from the database some day to keep it clean

That said, it's fine to set it to multiple hours. We will not implement "unlimited" though.

> A maximum number is also enforced for add-ons and a note on the maximum number that can be ordered is displayed in the frontend. This is unhelpful if the customer is theoretically allowed to order infinitely.

Well, the number of tickets per order is always constrained, never infinite, at least for technical reasons. For the add-ons, usually this is not about quotas at all but about business rules. But you opened a second discussion for it anyways, so no need to track that here

**Comment by DHD-Harper** [comment]
Thank you for your explanations.

But is it perhaps worth considering making the prominently placed countdown timer deactivatable? Even without a countdown timer, it is a typical experience for customers that a shopping cart will expire at some point. In our use case, there are no problems with vouchers and no seats are reserved.

> Prices are guaranteed until the cart expires, e.g. in case there is a price change while someone is buying

I have just tested this again: It is not the case.

If I add a product to the shopping cart and an automatic discount is automatically removed, I can no longer purchase the product at reduced price within the reservation period. During checkout, the price is automatically changed in the shopping cart.
